### PR TITLE
NSData -> Data

### PR DESCRIPTION
**Pull request by mhmiles**
Updates for swift 3.0/XCode8b6

### CODE REVIEW

**Comment by cezheng** [comment]
Why we need to change this? This PR is causing redundant copy of the whole data I guess.

**Comment by mhmiles** [comment]
My personal reason was that Alamofire requests now provide Data instead of NSData.  Should I pursue further?

**Comment by cezheng** [comment]
Hmm, what is annoying about the new Data struct is that it is a value type struct rather than a reference type class, which would be destructed when I goes out of scope, so it can not provide a getter to its underlying bytes, but rather, only copyBytes which is stupid for many cases.

https://developer.apple.com/reference/foundation/data

**Comment by cezheng** [comment]
Eh, forget what I've just said. It seems there is a method to access the underlying bytes:

https://developer.apple.com/reference/foundation/data/1780450-withunsafebytes

File: FuziTests/Resources/web.html [hunk 1]
```diff
@@ -851,7 +851,7 @@ <h1 itemscope itemtype="http://data-vocabulary.org/Breadcrumb" class="entry-titl
 
 <div class="highlight highlight-objective-c"><pre><span class="cp">#import "Ono.h"</span>
 
-<span class="n">NSData</span> <span class="o">*</span><span class="n">data</span> <span class="o">=</span> <span class="p">...;</span>
```
Comment (cezheng) [inline FuziTests/Resources/web.html]: Global replace?

File: FuziTests/AtomTests.swift [hunk 2]
```diff
@@ -26,9 +26,9 @@ class AtomTests: XCTestCase {
   var document: Fuzi.XMLDocument!
   override func setUp() {
     super.setUp()
-    let filePath = Bundle(for: AtomTests.self).path(forResource: "atom", ofType: "xml")!
```
Comment (cezheng) [inline FuziTests/AtomTests.swift]: guess this is not a filePath anymore, fileUrl? same applies to all other changes in tests.

File: FuziTests/AtomTests.swift [hunk 3]
```diff
@@ -26,9 +26,9 @@ class AtomTests: XCTestCase {
...
+    let filePath = Bundle(for: AtomTests.self).url(forResource: "atom", withExtension: "xml")!
```
Comment (cezheng) [inline FuziTests/AtomTests.swift]: guess this is not a filePath anymore, fileUrl? same applies to all other changes in tests.

File: Fuzi/Document.swift [hunk 4]
```diff
@@ -90,8 +90,8 @@ open class XMLDocument {
   
   - returns: An `XMLDocument` with the contents of the specified XML string.
   */
-  public convenience init(data: NSData) throws {
-    try self.init(cChars: [CChar](UnsafeBufferPointer(start: data.bytes.assumingMemoryBound(to: CChar.self), count: data.length)))
+  public convenience init(data: Data) throws {
+    try self.init(cChars: data.map({ CChar(bitPattern: $0) }))
```
Comment (cezheng) [inline Fuzi/Document.swift]: We should avoid copy here

https://developer.apple.com/reference/foundation/data/1780450-withunsafebytes

File: FuziTests/Resources/web.html (hunk 1 above)
Comment (mhmiles) [inline FuziTests/Resources/web.html]: Yes. Didn't know where this is used. Thought it might be correct

File: Fuzi/Document.swift (hunk 4 above)
Comment (mhmiles) [inline Fuzi/Document.swift]: Not sure there is any other way. The docs on withUnsafeBytes specifically say not to keep a reference outside of the scope. Couldn't come up with anything but a copy and figured the penalty isn't too big unless the xmldoc is huge.

Comment (cezheng) [inline Fuzi/Document.swift]: We don't keep reference on it, neither does the NSData approach. The data is just used for libxml2's parsing.

**Pull request closed without merging**